pkg/controller/backup_deletion_controller.go

```diff
@@ -442,6 +458,49 @@ func (c *backupDeletionController) deleteResticSnapshots(backup *v1.Backup) []er
 	return errs
 }
 
+func getCSIVolumeSnapshotsInBackup(b *velerov1api.Backup, csiClient snapshotter.SnapshotV1beta1Interface, log *logrus.Entry) ([]snapshotv1beta1api.VolumeSnapshot, []error) {
+	csiVolSnaps := []snapshotv1beta1api.VolumeSnapshot{}
+	errs := []error{}
+
+	selector := metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			velerov1api.BackupNameLabel: label.GetValidName(b.Name),
+		},
+	}
+	listOpts := metav1.ListOptions{
+		LabelSelector: metav1.FormatLabelSelector(&selector),
+	}
+
+	for _, ns := range backup.GetNamespacesInBackup(b) {
+		log.Debugf("Listing CSI volumesnapshots in namespace %s", ns)
+		nsvs, err := csiClient.VolumeSnapshots(ns).List(listOpts)
+		if err != nil {
+			errs = append(errs, err)
+		}
+		if nsvs.Items != nil && len(nsvs.Items) != 0 {
+			log.Debugf("Found %d CSI volumesnapshots in namespace %s", len(nsvs.Items), ns)
+			csiVolSnaps = append(csiVolSnaps, nsvs.Items...)
+		}
+	}
+
+	log.Infof("Found %d CSI volumesnapshots in backup %s", len(csiVolSnaps), b.Name)
+	return csiVolSnaps, errs
+}
+
+func (c *backupDeletionController) deleteCSIVolumeSnapshots(backup *velerov1api.Backup, csiClient snapshotter.SnapshotV1beta1Interface, log *logrus.Entry) []error {
```

Comment:
I think you should be able to simplify this function by using the `.DeleteCollection(...)` method and passing a label selector -- we use it elsewhere inside this controller for deleting related CRs in one API call.

Reply:
Considering that we need to `setVSCDeletionPolicyToDelete` on the associated volumesnapshotcontent, I don't think we will be able to the `DeleteCollection(..)` API. Let me know if I am missing something here.

Reply:
yeah, that sounds right. I wrote this comment before I thought about toggling the deletion policy.